lib/ts/recipe/passwordless/components/features/signInAndUp/index.tsx

```diff
@@ -38,23 +38,33 @@ import type { PasswordlessSignInUpAction, SignInUpState, SignInUpChildProps, Nor
 import type { RecipeInterface } from "supertokens-web-js/recipe/passwordless";
 import type { User } from "supertokens-web-js/types";
 
-export const useSuccessInAnotherTabChecker = (
-    state: SignInUpState,
-    dispatch: React.Dispatch<PasswordlessSignInUpAction>,
-    userContext: any
-) => {
+export const useRedirectAfterSuccess = (state: SignInUpState, recipeId: string, userContext: any, history: any) => {
     const callingConsumeCodeRef = useRef(false);
 
     useEffect(() => {
         // We only need to start checking this if we have an active login attempt
-        if (state.loginAttemptInfo && !state.successInAnotherTab) {
+        if (state.loginAttemptInfo) {
             const checkSessionIntervalHandle = setInterval(async () => {
                 if (callingConsumeCodeRef.current === false) {
                     const hasSession = await Session.doesSessionExist({
                         userContext,
                     });
                     if (hasSession) {
-                        dispatch({ type: "successInAnotherTab" });
+                        clearInterval(checkSessionIntervalHandle);
+                        void SessionRecipe.getInstanceOrThrow().validateGlobalClaimsAndHandleSuccessRedirection(
+                            {
+                                rid: recipeId,
+                                successRedirectContext: {
+                                    action: "SUCCESS",
+                                    isNewRecipeUser: false,
+                                    user: undefined,
```

Comment:
Sort of related to this is that the [type of SUCCESS action](https://github.com/supertokens/supertokens-auth-react/blob/master/lib/ts/recipe/authRecipe/types.ts#L29), is strange. The `user` obj can be `undefined` in case of `isNewRecipeUser: false`. This leads to a bad developer experience. 

If there is no place where user is actually undefined (other than this place), and if we remove this "cronjob" then w should change the type of `GetRedirectionURLContext` like:
```
{
          action: "SUCCESS";
          isNewRecipeUser: boolean;
          user: User;
          redirectToPath?: string;
      }
```

Reply:
https://github.com/supertokens/supertokens-auth-react/pull/766#discussion_r1413946449